systemd/escape.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package systemd
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+const allowed = `:_.abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789`
+
+// EscapePath works like systemd-escape --path
+// FIXME: use github.com/coreos/go-systemd/unit/escape.go
```

Comment:
I wouldn't bother unless there are other benefits to bringing this in. The logic below looks small and contained enough that avoiding the additional dependency seems good.

One idea: `EscapePath` isn't actually escaping a path. It's escaping a unit name, with special handling for a path that starts with `"/"`. So maybe `EscapeUnitName` is a better term here, and we can still document that it behaves like `systemd-escape --path` as done above